src/test/java/com/nvidia/spark/rapids/jni/RmmSparkTest.java

```diff
@@ -799,4 +845,53 @@ public boolean onAllocFailure(long sizeRequested, int retryCount) {
       return false;
     }
   }
+
+  private static class AllocatingRmmEventHandler extends BaseRmmEventHandler {
+    // if true, we are still in the onAllocFailure callback (recursive call)
+    boolean stillHandlingAllocFailure = false;
+
+    int allocationCount;
+
+    long allocSize;
+
+    public int getAllocationCount() {
+      return allocationCount;
+    }
+
+    public AllocatingRmmEventHandler(long allocSize) {
+      this.allocSize = allocSize;
+    }
+
+    @Override
+    public boolean onAllocFailure(long sizeRequested, int retryCount) {
+      // Catch all JVM exceptions in this handlers to prevent loops:
+      // - If we throw a Java exception here, it will get caught in the RmmJni on_alloc_fail
+      //   method. This causes a native std::runtime_error to be thrown.
+      // - The std::runtime_error is then caught in spark_resource_adaptor::do_allocate and
```

Comment:
Isn't the issue that we retry on any exception that is not an OOM exception? I am not sure why we would want to retry for other exception types. That feels wrong to me.  And I think would be a lot better than catching a Throwable and turning it into a boolean that loses all of the information about it.

Reply:
I can remove special handling of `std::exception` here: https://github.com/NVIDIA/spark-rapids-jni/pull/1122/files#diff-936903e8604f063e1364ecc9ad83dcd4d5552d6c26d93073b469f99ca562d7a3R1171 or perhaps leave the handling but always throw (instead of conditionally). Does that make sense? Or should it be removed entirely.